superset/datasets/api.py

```diff
@@ -123,6 +125,36 @@ class DatasetRestApi(BaseSupersetModelRestApi):
         "schema",
         "sql",
         "table_name",
+        "always_filter_main_dttm",
```

Comment:
@dpgaspar won't the payload critically increase with this change?

Reply:
We have datasets with thousands of metrics and columns.

Reply:
If it's only blocker, I don't mind to revert that. Perhaps it makes more sense to introduce a parallel API (v2) and catch up the front-end later. Please let me know, no rush here.

Reply:
Could potentially generate huge payloads like @michael-s-molina is saying

Reply:
Looks like this was reverted...